### PR TITLE
Implement automatic backups

### PR DESCRIPTION
**Pull request by Naveen3Singh**
This PR addresses https://github.com/SimpleMobileTools/Simple-Calendar/issues/174

This is very similar to the existing "Export event" feature but it is automatic and happens every day.

### Notes:
 - Backups are executed at 6 AM every day (hardcoded)
 - In case the device was off at the scheduled time, backup happens whenever the device turns on.
 - This feature is only available on Android 11 and above (for now)
 - `QuickFilterEventTypeDialog` was removed as it was basically a duplicated version of the `SelectEventTypeDialog`.

### CODE REVIEW

**Comment by Naveen3Singh** [comment]
let me know if you want to show a notification when exporting fails.

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt [hunk 1]
```diff
@@ -526,7 +526,11 @@ class MainActivity : SimpleActivity(), RefreshRecyclerViewListener {
     }
 
     private fun showFilterDialog() {
-        FilterEventTypesDialog(this) {
+        SelectEventTypesDialog(this, config.displayEventTypes) {
+            if (config.displayEventTypes != it) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/MainActivity.kt:530]: I think we can skip the check here, or maybe add a "return" if nothing changed, depending on the desired behaviour

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/SettingsActivity.kt [hunk 2]
```diff
@@ -830,6 +838,42 @@ class SettingsActivity : SimpleActivity() {
...
+        settings_backups_label.beVisibleIf(isRPlus())
+        settings_backups_divider.beVisibleIf(isRPlus())
+        settings_enable_automatic_backups_holder.beVisibleIf(isRPlus())
+        settings_enable_automatic_backups.isChecked = config.autoBackup
+        settings_enable_automatic_backups_holder.setOnClickListener {
+            val wasBackupDisabled = !config.autoBackup
+            if (wasBackupDisabled) {
+                ManageAutomaticBackupsDialog(
+                    activity = this,
+                    onSuccess = { scheduleNextAutomaticBackup() },
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/SettingsActivity.kt:851]: add these onto a newline like {
scheduleNextAutomaticBackup()
}

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/receivers/BootCompletedReceiver.kt [hunk 3]
```diff
@@ -16,6 +17,13 @@ class BootCompletedReceiver : BroadcastReceiver() {
                 scheduleAllEvents()
                 notifyRunningEvents()
                 recheckCalDAVCalendars(true) {}
+                scheduleNextAutomaticBackup()
+                val now = getNowSeconds()
+                val intervalInSeconds = AUTO_BACKUP_INTERVAL_IN_DAYS * DAY
+                if (config.autoBackup && config.lastAutoBackupTime !in (now - intervalInSeconds)..now) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/receivers/BootCompletedReceiver.kt:23]: that last part is too confusing, cant we make it more readable?

**Comment by Naveen3Singh** [comment]
Okay, check again please

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/dialogs/ManageAutomaticBackupsDialog.kt [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                if (backupFolder.isEmpty()) {
+                    activity.getString(R.string.select_folder)
+                } else {
+                    activity.humanizePath(backupFolder)
+                }
+            )
+
+            val filename = config.autoBackupFilename.ifEmpty {
+                "${activity.getString(R.string.events)}_%Y%M%D_%h%m%s"
+            }
```
[the remark this answers is not in the document]
Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/dialogs/ManageAutomaticBackupsDialog.kt:31]: Newlines are needed at multiple places below aswell

[the remark this answers is not in the document]
Comment (Naveen3Singh) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/dialogs/ManageAutomaticBackupsDialog.kt:31]: ah, understood.

[the remark this answers is not in the document]
Comment (Naveen3Singh) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/dialogs/ManageAutomaticBackupsDialog.kt:31]: done :+1:

**Comment by tibbi** [comment]
toggle the "Enable automatic backups" in the app settings only after confirming the dialog with OK, not right at pressing

**Comment by tibbi** [comment]
lets set some default folder for saving the files, like Downloads or so, do not show "Select a folder" there. Also rename the top label from "Select a folder" to "Folder"

**Comment by tibbi** [comment]
also lets activate all 3 checkboxes by default

**Comment by Naveen3Singh** [comment]
> lets set some default folder for saving the files, like Downloads or so

saving it to Downloads/Events as Downloads may not be accessible

**Comment by tibbi** [comment]
make it just Downloads, no need for any subfolder. Check again how is the voice recorder done

**Comment by waleedsamy** [comment]
> This feature is only available on Android 11 and above (for now)

Hope you extend it to support Android 10 soon!

**Pull request merged**